### ISSUE TITLE
Are Convolutional Neural Networks or Transformers more like human vision?

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#59]
## Motivation

- ImageNet trained CNNs are biased towards texture. How about vision transformers? Which are more aligned with human perception?

## Key Idea

- Cluster ImageNet labels to 16 groups according to Word hierarchy and construct confusion matrix
- Study shape/texture bias as well as error consistency (using confusion matrix JS divergence)

## Previous Approach

- Stylized ImageNet dataset was used to disentangle texture from shape, and showed that CNNs are biased towards texture.
- Finetuning on this diagnostic dataset helps increase shape bias.

## Method

- Group labels into 16 semantic categories, and construct a confusion matrix for each model (ViT and ResNet)
- Compute Cohen's Kappa, Class-wise and inter-class JS divergence between the above model and human decision maker.

- Cohen's Kappa measures decision overlaps (both right and wrong) and adjusts for the overall accuracy effect (Higher accuracy leads to larger overlap)
![image](https://user-images.githubusercontent.com/46308407/118753359-ccc97e00-b89f-11eb-81af-3790eb314eb3.png)

<br>

- Class-wise JS divergence compares classwise error distribution (16 dim vectors)
- Inter-class JS divergence measures off-diagonal elements in confusion matrix (i.e. what is misclassified as what)

<br>

- Shape bias is computed by: {# shape is correct} / ({# shape is correct} + {# texture is correct})

## Result

![image](https://user-images.githubusercontent.com/46308407/118753700-7dd01880-b8a0-11eb-81d3-7ff0ac42e253.png)

- Class-wise divergence and Cohen's Kappa favors transformers, while inter-class divergence favors ResNet.
- Class-wise divergence with human perception is a good indicator for accuracy on Stylized Image Net. (That is, biased towards shapes)

![image](https://user-images.githubusercontent.com/46308407/118753836-bf60c380-b8a0-11eb-8ebb-f227e0ccf547.png)

- Again, inter-class divergence favors CNNs over transformers.

![image](https://user-images.githubusercontent.com/46308407/118753973-00f16e80-b8a1-11eb-8f84-e1a39dc7f561.png)

- Transformers are more biased towards shape than CNNs, which aligns with human perception.

![image](https://user-images.githubusercontent.com/46308407/118754007-15ce0200-b8a1-11eb-9883-7d6bd7c5d7fa.png)

- Data augmentations like color distortion, blurring, and noising is known to help models pay more attention to shape (as they destroy shape clues)

![image](https://user-images.githubusercontent.com/46308407/118754054-2da58600-b8a1-11eb-8098-e28f230e009c.png)

## Comment

- Good topic, and this kind of *finding* papers are always insightful.
- However, this particular paper had rather confusing intuitions. We can see that transformers favor shape clues over texture clues compared to CNNs (which is quite obvious), but their error consistency analysis doesn't give clear conclusions on which is better than which.
- Plus, the fact that data augmentation reduces CNN's similarity with human decision making could be seen as an implication that their metric, namely the inter-class JS divergence doesn't accurately reflect similarity in decision making. It could be noisy (due to rather high dimension), or as they've pointed out, their can be confounding effects such as conceptual hierarchy. (Dogs are more similar to cats than trucks, though this is not reflected in the metric)
- As mentioned in the paper, conceptual hierarchy indeed seems to be an important factor in human perception, so further study is in need. What about making the clusters even more coarse, as in: mammals, fish, birds, vehicles, ...? (Like less than 10)
